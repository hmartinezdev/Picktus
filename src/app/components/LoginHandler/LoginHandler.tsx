import Button from '@components/Button';
import Input from '@components/Input';
import SubscribeHandler from '@components/SubscribeHandler';
import colors from '@constants/colors';
import React, { Component } from 'react';

class LoginHandler extends Component {
  public render(): React.ReactElement<LoginHandler> {
    return (
      <div className="container">
        <SubscribeHandler />
        <div className="login">
          <Input placeholder="Mail" />
          <Input placeholder="Password" type="password" />
          <Button text="Login" />
        </div>
        <style jsx>{`
          .container {
            position: relative;
          }

          .login {
            display: flex;
            flex-direction: column;
            align-items: center;
            background-color: ${colors.primary};
            min-width: 22rem;
            padding: 1.5rem;
            box-sizing: border-box;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }
        `}</style>
      </div>
    );
  }
}
export default LoginHandler;
