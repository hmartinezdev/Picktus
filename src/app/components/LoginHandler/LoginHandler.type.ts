export interface IStringMap {
  [s: string]: string;
}
export interface ILoginHandlerState {
  inputs: IStringMap;
}
