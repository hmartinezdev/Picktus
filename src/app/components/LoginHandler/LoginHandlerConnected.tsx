import { classicLogin, facebookLogin, googleLogin, twitterLogin } from '@store/reducers/user/user-actions';
import { connect } from 'react-redux';
import LoginHandler from './LoginHandler';

export default connect(
  undefined,
  { classicLogin, googleLogin, facebookLogin, twitterLogin }
)(LoginHandler);
