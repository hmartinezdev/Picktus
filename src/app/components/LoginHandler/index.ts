import LoginHandler from './LoginHandler';

export default LoginHandler;
