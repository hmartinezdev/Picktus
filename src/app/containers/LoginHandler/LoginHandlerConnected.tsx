import { signin } from '@store/reducers/user/user-actions';
import { connect } from 'react-redux';
import LoginHandler from './LoginHandler';

export default connect(
  undefined,
  { signin }
)(LoginHandler);
