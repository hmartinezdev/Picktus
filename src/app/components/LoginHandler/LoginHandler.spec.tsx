import enzyme, { shallow } from 'enzyme';
import React from 'react';
import LoginHandler from './LoginHandler';

let wrapper: enzyme.ShallowWrapper;

const setup = (props = {}) => shallow(<LoginHandler {...props} />);

describe('<LoginHandler />', () => {
  beforeEach(() => {
    wrapper = setup();
  });

  test('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
