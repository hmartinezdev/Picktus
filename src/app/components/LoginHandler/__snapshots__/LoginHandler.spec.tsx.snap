// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginHandler /> should render properly 1`] = `
<div
  className="container"
>
  <SubscribeHandler />
  <div
    className="login"
  >
    <Input
      placeholder="Mail"
      type="text"
    />
    <Input
      placeholder="Password"
      type="password"
    />
    <Button
      onClick={[Function]}
      text="Login"
    />
  </div>
  <style
    jsx={true}
  >
    
          .container {
            position: relative;
          }

          .login {
            display: flex;
            flex-direction: column;
            align-items: center;
            background-color: #2d3047cd;
            min-width: 22rem;
            padding: 1.5rem;
            box-sizing: border-box;
            border-radius: 3px;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
          }
        
  </style>
</div>
`;
